service/gcs/errors/errors.go

```diff
@@ -45,3 +62,111 @@ func (e *processDoesNotExistError) Error() string {
 func NewProcessDoesNotExistError(pid int) *processDoesNotExistError {
 	return &processDoesNotExistError{Pid: pid}
 }
+
+// StackTracer is an interface originating (but not exported) from the
+// github.com/pkg/errors package. It defines something which can return a stack
+// trace.
+type StackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+type baseHresultError struct {
+	hresult Hresult
+}
+
+func (e *baseHresultError) Error() string {
+	return fmt.Sprintf("HRESULT: %d", e.Hresult())
```

Comment:
It would be nice to use %x and cast e.Hresult() to a uint32 so that it looks useful (e.g. 0x8000FFFF instead of -248w49834whatever).